### PR TITLE
fis-64-refactor-graphql-setup

### PR DESCRIPTION
**Pull request by bienmag**
all graphql should be refactored to using apollo client.
all the routes should be working fine.
i also added get 'codegen' files which generates the types for us. 
so it would warn about an error when generating the types 

and it will help when the data is fetched 
image
the files could be run with yarn compile and i think should be run when you make changes in the schema, resolvers or queries

also i separated queries and mutations in the server and added generated types for them

### CODE REVIEW

**Comment by craigostrin** [comment]
in the next daily sync, could you give an overview of how codegen works? and talk a bit about the need for 'yarn compile'?

**Comment by craigostrin** [comment]
please add the steps we need to take to test this branch besides reviewing the code

File: apps/server/src/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,100 @@
+  type Query {
```
[the remark this answers is not in the document]
Comment (bienmag) [reply apps/server/src/schema.graphql]: done

File: apps/client/lib/apollo-wrapper.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  ApolloNextAppProvider,
+  NextSSRInMemoryCache,
+  NextSSRApolloClient,
+  SSRMultipartLink,
+} from '@apollo/experimental-nextjs-app-support/ssr'
+
+// have a function to create a client for you
+function makeClient() {
+  const httpLink = new HttpLink({
+    uri: process.env.NEXT_PUBLIC_GRAPHQL_API_URL,
```
[the remark this answers is not in the document]
Comment (bienmag) [reply apps/client/lib/apollo-wrapper.js:14]: done

File: apps/client/app/employee/[id]/new-review/metric-list.tsx [hunk 3]
```diff
@@ -2,29 +2,12 @@
 
 import { Button } from '@/components/ui/button'
 import Metric from '@/components/ui/metric'
-import { updateReport } from '@/lib/fetch'
+
 import { GradeData, ReportData } from '@/types/models'
```
[the remark this answers is not in the document]
Comment (bienmag) [reply apps/client/app/employee/[id]/new-review/metric-list.tsx]: done

File: apps/client/app/dashboard/page.tsx [hunk 4]
```diff
@@ -1,54 +1,33 @@
...
-import { useEffect, useState } from 'react'
-import axios from 'axios'
 import '@/app/global.css'
 import Link from 'next/link'
 import { Button } from '@/components/ui/button'
-import { getCookie, setCookie } from 'cookies-next'
+import { cookies } from 'next/headers'
 import UserItem from '@/components/table/UserItem'
+import { GET_USERS } from '@/lib/queries'
+import { UserModel } from '../../../../packages/types/models'
```
[the remark this answers is not in the document]
Comment (bienmag) [reply apps/client/app/dashboard/page.tsx]: done

**Review by craigostrin — APPROVED** [review]

**Pull request merged**